### PR TITLE
feat(pipettes): initial pipette sensor board bringup

### PR DESCRIPTION
**Pull request by Laura-Danielle**
## Overview
Initial work to bring-up all of the sensors available on the pipette sensor board. Additional work is needed for bringing up a sensor task and sending/receiving messages through CAN Bus. Those tickets will be added to Jira shortly.

Closes RET-46 RET-47 and RET-48

### CODE REVIEW

File: pipettes/firmware/i2c.c [hunk 1]
```diff
@@ -53,3 +112,65 @@ I2C_HandleTypeDef* MX_I2C_Init()
...
+
+void enable_eeprom() {
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+}
+
+void disable_eeprom() {
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+}
+
+I2CHandlerStruct i2c_setup(PipetteType pipette_type) {
```
Comment (amitlissack) [inline pipettes/firmware/i2c.c:164]: It is very un-C like to pass or return structs by value. 

I realize this function is only called once so this isn't a big deal, but for the purpose of writing in the C idiom I suggest you rewrite this function to accept a pointer to a `I2CHandlerStruct`.

Comment (Laura-Danielle) [reply pipettes/firmware/i2c.c:164]: That makes sense! Generally, is there anything you would want to return by value aside from a primitive type?

Comment (amitlissack) [reply pipettes/firmware/i2c.c:164]: In C, just primitives.

Comment (sfoster1) [reply pipettes/firmware/i2c.c:164]: You're right but also that's a thing I've always hated about C, and it's kind of held over from older implementations and standards. I'm ok with either way.

**Review by sfoster1 — APPROVED** [review]

**Pull request merged**